engine/src/execution_graph/logic_controllers/CacheMachine.cpp

```diff
@@ -686,6 +683,41 @@ bool CacheMachine::has_messages_now(std::vector<std::string> messages){
 	return true;
 }
 
+std::unique_ptr<ral::cache::CacheData> CacheMachine::pullAnyCacheData(const std::vector<std::string> & messages) {
```

Comment:
we are starting to accumulate way oto many ways of pulling data from these caches. I think its time we consolidate and have some kind of flag to handle all these different use cases

Reply:
do you have any suggestions? i am not sure what is a better alternative